account/src/main/java/com/caldevsupplychain/account/controller/AccountControllerImpl.java

```diff
@@ -84,24 +86,161 @@
+		return new ResponseEntity<>(userMapper.toWS(user.get()), HttpStatus.OK);
 	}
 
+	@PostMapping("/login")
+	public ResponseEntity<?> login(@Validated @RequestBody UserWS userWS) {
+
+		Optional<UserBean> user = accountService.findByEmailAddress(userWS.getEmailAddress());
+
+		if (!user.isPresent()) {
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.LOGIN_INVALID.name(), "Invalid Login"), HttpStatus.UNAUTHORIZED);
+		}
+
+		Subject subject = SecurityUtils.getSubject();
+
+		// Apache Shiro authentication check for 1st login
+		if (!subject.isAuthenticated()) {
+			UsernamePasswordToken token = new UsernamePasswordToken(userWS.getEmailAddress(), userWS.getPassword());
+
+			try {
+				subject.login(token);
+			} catch (AuthenticationException e) {
+				log.error("Error in login. User fail to login. user={}", user.toString());
+				return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.LOGIN_INVALID.name(), e.getMessage()), HttpStatus.UNAUTHORIZED);
+			}
+		}
+
+		log.info("Success in user login. user={}", user.toString());
+
+		UserBean userBean = user.get();
+		JwtBean jwtBean = jwtMapper.toBean(userBean);
+		JWTAuthenticationToken jwtObj = jwtService.createJwtToken(jwtBean);
+
+		log.info("Success create jwt token. jwtToken={}", jwtObj.getToken());
+
+		return new ResponseEntity<>(userMapper.toWS(userBean), jwtService.createJwtHeader(jwtObj.getToken()), HttpStatus.OK);
+	}
+
+
+	@GetMapping("/logout")
+	@RequiresAuthentication
+	public ResponseEntity<?> logout() {
+		Subject subject = SecurityUtils.getSubject();
+
+		Optional<UserBean> user = accountService.findByUuid(subject.getPrincipal().toString());
+
+		if (!user.isPresent()) {
+			log.error("Error in logout. User fail to logout. user={}", user.toString());
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.LOGOUT_INVALID.name(), "Logout Invalid"), HttpStatus.BAD_REQUEST);
+		}
+
+		log.info("Success in user logout. user={}", user.toString());
+
+		subject.logout();
+
+		return new ResponseEntity<>(userMapper.toWS(user.get()), HttpStatus.OK);
+	}
+
+	@PostMapping("/users/{uuid}/forgot_password")
+	public ResponseEntity<?> forgotPassword(@PathVariable String uuid, @Validated @RequestBody UserWS userWS) {
+
+		Optional<UserBean> user = accountService.findByUuid(uuid);
+
+		if (!user.isPresent()) {
+			log.error("Error in user forgot password. Fail in finding user's uuid={}", uuid);
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ACCOUNT_NOT_EXIST.name(), "Cannot find user account."), HttpStatus.NOT_FOUND);
+		}
+
+		if(!userWS.getEmailAddress().equals(user.get().getEmailAddress())){
+			log.error("Error in user forgot password. Fail in finding user's email={}", userWS.getEmailAddress());
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ACCOUNT_NOT_EXIST.name(), "Cannot find user email address."), HttpStatus.NOT_FOUND);
+		}
+
+		UserBean userBean = user.get();
+
+		userBean = accountService.setPassCode(userBean);
```

Comment:
You can reuse `accountService.updateUser()` instead of creating a new method.

Reply:
fix